File: app/src/main/java/com/banzo/inaction/inaction/ResetSenha.java
package com.banzo.inaction.inaction;

import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.FirebaseAuth;

public class ResetSenha extends AppCompatActivity {
    private EditText editResetEmail;
    private Button btnResetSenha;

    private FirebaseAuth auth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_reset_senha);
        iniciarComponentes();
        eventoClick();
    }

    private void eventoClick() {
        btnResetSenha.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String email = editResetEmail.getText().toString().trim();
                resetSenha(email);
            }
        });
    }

    private void resetSenha(String email) {
        auth.sendPasswordResetEmail(email)
                .addOnCompleteListener(ResetSenha.this, new OnCompleteListener<Void>() {
                    @Override
                    public void onComplete(@NonNull Task<Void> task) {
                        if (task.isSuccessful()){
                            alert("Um email foi enviado para alterar a senha");
                            finish();
                        }else{
                            alert("Email não encontrado");
                        }
                    }
                });
    }

    private void alert(String s) {
        Toast.makeText(ResetSenha.this, s, Toast.LENGTH_SHORT).show();
    }

    private void iniciarComponentes() {
        editResetEmail = findViewById(R.id.editResetEmail);
        btnResetSenha = findViewById(R.id.btnResetSenha);

    }

    @Override
    protected void onStart() {
        super.onStart();
        auth = Conexao.getFirebaseAuth();
    }
}

File: app/src/main/java/com/banzo/inaction/inaction/ResetSenha.kt
package com.banzo.inaction.inaction

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import com.google.firebase.auth.FirebaseAuth

class ResetSenha : AppCompatActivity() {
    private var editResetEmail: EditText? = null
    private var btnResetSenha: Button? = null

    private var auth: FirebaseAuth? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_reset_senha)
        iniciarComponentes()
        eventoClick()
    }

    private fun eventoClick() {
        btnResetSenha!!.setOnClickListener {
            val email = editResetEmail!!.text.toString().trim { it <= ' ' }
            resetSenha(email)
        }
    }

    private fun resetSenha(email: String) {
        auth!!.sendPasswordResetEmail(email)
                .addOnCompleteListener(this@ResetSenha) { task ->
                    if (task.isSuccessful) {
                        alert("Um email foi enviado para alterar a senha")
                        finish()
                    } else {
                        alert("Email não encontrado")
                    }
                }
    }

    private fun alert(s: String) {
        Toast.makeText(this@ResetSenha, s, Toast.LENGTH_SHORT).show()
    }

    private fun iniciarComponentes() {
        editResetEmail = findViewById(R.id.editResetEmail)
        btnResetSenha = findViewById(R.id.btnResetSenha)

    }

    override fun onStart() {
        super.onStart()
        auth = Conexao.getFirebaseAuth()
    }
}
